File: front-end/src/app/service-models/app/v1/client/client-employee-leave-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';
import { LeaveTypeSM } from '../../enums/leave-type-s-m.enum';

export class ClientEmployeeLeaveSM extends CoinManagementServiceModelBase<number> {
    clientUserId!: number;
    clientCompanyDetailId!: number;
    leaveType!: LeaveTypeSM;
    employeeComment!: string;
    isApproved?: boolean;
    approvedByUserName!: string;
    approvalComment!: string;
    leaveDateFromUTC!: Date;
    leaveDateToUTC!: Date;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/ClientEmployeeLeaveSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;
using Siffrum.Web.Payroll.ServiceModels.Enums;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class ClientEmployeeLeaveSM : SiffrumPayrollServiceModelBase<int>
    {
        public int ClientUserId { get; set; }
        public int ClientCompanyDetailId { get; set; }
        public LeaveTypeSM LeaveType { get; set; }
        public string EmployeeComment { get; set; }
        public bool? IsApproved { get; set; }
        public string ApprovedByUserName { get; set; }
        public string ApprovalComment { get; set; }
        public DateTime LeaveDateFromUTC { get; set; }
        public DateTime LeaveDateToUTC { get; set; }
    }
}
